gprt/gprt.cpp

```diff
@@ -1151,46 +1151,63 @@ struct TriangleAccel : public Accel {
       &accelerationStructureBuildSizesInfo
     );
     
-    accelBuffer = new Buffer(
-      physicalDevice, logicalDevice, VK_NULL_HANDLE, VK_NULL_HANDLE, 
-      // means we can use this buffer as a means of storing an acceleration structure
-      VK_BUFFER_USAGE_ACCELERATION_STRUCTURE_STORAGE_BIT_KHR | 
-      // means we can get this buffer's address with vkGetBufferDeviceAddress
-      VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT, 
-      // means that this memory is stored directly on the device 
-      //  (rather than the host, or in a special host/device section)
-      VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT,
-      accelerationStructureBuildSizesInfo.accelerationStructureSize
-    );
-
-    scratchBuffer = new Buffer(
-      physicalDevice, logicalDevice, VK_NULL_HANDLE, VK_NULL_HANDLE, 
-      // means that the buffer can be used in a VkDescriptorBufferInfo. // Is this required? If not, remove this...
-      VK_BUFFER_USAGE_STORAGE_BUFFER_BIT | 
-      // means we can get this buffer's address with vkGetBufferDeviceAddress
-      VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT, 
-      // means that this memory is stored directly on the device 
-      //  (rather than the host, or in a special host/device section)
-      VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT,
-      accelerationStructureBuildSizesInfo.buildScratchSize
-    );
-
-    VkAccelerationStructureCreateInfoKHR accelerationStructureCreateInfo{};
-    accelerationStructureCreateInfo.sType = VK_STRUCTURE_TYPE_ACCELERATION_STRUCTURE_CREATE_INFO_KHR;
-    accelerationStructureCreateInfo.buffer = accelBuffer->buffer;
-    accelerationStructureCreateInfo.size = accelerationStructureBuildSizesInfo.accelerationStructureSize;
-    accelerationStructureCreateInfo.type = VK_ACCELERATION_STRUCTURE_TYPE_BOTTOM_LEVEL_KHR;
-    err = gprt::vkCreateAccelerationStructure(
-      logicalDevice,
-      &accelerationStructureCreateInfo, 
-      nullptr,
-      &accelerationStructure
-    );
-    if (err) GPRT_RAISE("failed to create acceleration structure for triangle accel build! : \n" + errorString(err));
+    if (accelBuffer && accelBuffer->size != accelerationStructureBuildSizesInfo.accelerationStructureSize)
+    {
+      // Destroy old accel handle too
+      gprt::vkDestroyAccelerationStructure(logicalDevice, accelerationStructure, nullptr);
+      accelerationStructure = VK_NULL_HANDLE;
+      accelBuffer->destroy();
+      delete(accelBuffer);
```

Comment:
Does it make sense to put a `destroy()` call into `~Buffer`? 

Then this can simply be `delete(accelBuffer)`.

Reply:
This could also be done in a subsequent PR.